components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.ts

```diff
@@ -1,139 +1,80 @@
-  // The map of ProjectChecked by id. Any checked changes propagated via
-  // onProjectChecked. Updates should be applied to parent component state.
-  @Input() projects: ProjectCheckedMap = {};
-
-  // Setting disabled to true means the dropdown will be unusable and will have a grey background
-  @Input() disabled = false;
+export class ProjectsDropdownComponent implements OnInit, OnDestroy, OnChanges {
 
   // Used to re-synchronize summary label if the set of checked items has changed.
   // This optional input is needed only when re-displaying the project dropdown
   // for *additional* resources, as with the create-object-modal-component.
   // Other consumers, e.g. team-details.component use it only for a single resource.
   @Input() projectsUpdated: EventEmitter<boolean>;
 
-  // Emits a project that changed as a result of a check or uncheck.
-  @Output() onProjectChecked = new EventEmitter<ProjectChecked>();
+  @Input() checkedProjectIDs: string[] = [];
 
-  // filteredProjects is merely a container to hold the projectsArray
-  // that can be altered
-  public filteredProjects: ProjectChecked[] = [];
-  public active = false;
-  public label = UNASSIGNED_PROJECT_ID;
-  public filterValue = '';
-
-  ngOnInit(): void {
-    if (this.projectsUpdated) { // an optional setting
-      this.projectsUpdated.subscribe(() => {
-        this.updateLabel();
-      });
-    }
-  }
-
-  ngOnChanges(changes: SimpleChanges): void {
-    this.updateLabel();
-    // only update the projects list on initialization/the first change
-    if (changes.projects && changes.projects.firstChange) {
-      this.filteredProjects = this.projectsArray;
-    }
-  }
+  // Emits checked set of ids upon completion.
+  @Output() onDropdownClosing = new EventEmitter<string[]>();
 
-  get projectsArray(): ProjectChecked[] {
-    return ChefSorters.naturalSort(Object.values(this.projects), 'name');
-  }
+  // Label to use when none are selected
+  public noneSelectedLabel = ProjectConstants.UNASSIGNED_PROJECT_ID;
 
-  toggleDropdown(event: MouseEvent): void {
-    event.stopPropagation();
-    if (this.disabled) {
-      return;
-    }
-    if (!this.active) {
-      this.filterValue = '';
-      this.filteredProjects = this.projectsArray;
-    }
+  // plural display name of resource
+  public objectNounPlural = 'projects';
 
-    this.active = !this.active;
-  }
+  public projects: ResourceChecked[];
 
-  projectChecked(checked: boolean, project: ProjectChecked): void {
-    project.checked = checked;
-    this.updateLabel();
-    this.onProjectChecked.emit(project);
-  }
+  private isDestroyed = new Subject<boolean>();
 
-  closeDropdown(): void {
-    if (this.active) {
-      this.active = false;
-    }
-  }
+  constructor(private store: Store<NgrxStateAtom>) { }
 
-  handleFilterKeyUp(): void {
-    this.filteredProjects = this.filterProjects(this.filterValue);
+  ngOnInit(): void {
+    this.store.select(assignableProjects)
```

Comment:
This block of code to generate the projects list from the ngrx store was previously duplicated by every caller; now it is encapsulated here, in the component itself.